ush/python/pygfs/task/globus_hpss.py

```diff
@@ -0,0 +1,422 @@
+        while not all(transfer_set["completed"]) and wait_count < max_wait_count:
+            sleep(sleep_time)
+            for i in range(len(transfer_set["locations"])):
+                status_file = transfer_set["status_files"][i]
+                if os.path.exists(status_file):
+                    # If this is a new status file, check if the transfer was successful
+                    if not transfer_set["completed"][i]:
+                        transfer_set["completed"][i] = True
+                        with open(status_file) as status_handle:
+                            status_string = status_handle.readline().rstrip()
+                            transfer_set["successes"][i] = status_string == f"status.{transfer_set['xfer_ids'][i]} SUCCESS"
+
+                        if transfer_set["successes"][i]:
+                            logger.info(f"Successfully archived {transfer_set['locations'][i]} to HPSS!")
+                        else:
+                            # Exit the loop immediately, but allow the log file to be downloaded.
+                            if has_rstprod:
+                                logger.error(
+                                    f"FATAL ERROR HPSS archiving of restricted file {transfer_set['locations'][i]} failed!"
+                                    "\nPlease verify that the file has been deleted from HPSS!"
+                                )
+                                transfer_failed = True
+                                break
+                            else:
+                                logger.error(f"FATAL ERROR HPSS archiving failed for {transfer_set['locations'][i]}.")
+                                transfer_failed = True
+
+            # Retrieve the log file (if it exists) from the server and check if it failed
+            try:
+                self.scp(f"{server_name}:{server_job_dir}/run_doorman.log", '.')
+            except ProcessError:
+                check_log_count += 1
+                if check_log_count > 3:
+                    logger.error(f"FATAL ERROR Unable to retrieve the run_doorman.log file")
+                    transfer_failed = True
+            else:
+                with open("run_doorman.log") as doorman_log:
+                    doorman_lines = doorman_log.readlines()
+
+                log_read = True
+
+                if "FAILURE" in doorman_lines[-1]:
+                    logger.error(f"FATAL ERROR The doorman failed to run on {server_name}")
+                    transfer_failed = True
+
+            if transfer_failed:
+                break
+
+            wait_count += 1
+            wait_time = wait_count * sleep_time
+
+            complete_count = sum(transfer_set["completed"])
+
+            logger.debug(f"{complete_count} files transferred in {wait_time} seconds.")
+
+        # Sleep a couple more seconds to ensure all status files finish transferring
+        sleep(2)
```

Comment:
might want to bump this up to 5 just to be safe?

Reply:
Good idea.  Done.